vicodyn/include/cocaine/vicodyn/balancer/utils.hpp

```diff
@@ -0,0 +1,25 @@
+#include <cstdlib>
+#include <vector>
+
+namespace cocaine {
+namespace vicodyn {
+namespace balancer {
+
+template<class Iterator, class Predicate>
+auto choose_random_if(Iterator begin, Iterator end, std::size_t size, Predicate p) -> Iterator {
+    std::vector<Iterator> chosen;
+    chosen.reserve(size);
+    for(; begin != end; ++begin) {
+        if(p(*begin)) {
```

Comment:
Is it assumed that `p` predicate could change an item? If not, may be express immutability via constant iterators as function parameters?

Reply:
Yes, but methods in STL with similar behavior allows to change an items too. Ex: remove_if, find_if